src/cpp/session/resources/grid/dtviewer.js

```diff
@@ -150,15 +150,25 @@ var highlightSearchMatch = function(data, search, pos) {
 // render cell contents--if no search is active, just renders the data
 // literally; when search is active, highlights the portion of the text that
 // matches the search
-var renderCellContents = function(data, type, row, meta) {
+var renderCellContents = function(data, type, row, meta, clazz) {
 
   // usually data is a string; 0 is a special value signifying NA 
   if (data === 0 || (displayNullsAsNAs && data === null)) {
     return '<span class="naCell">NA</span>';
   }
 
-  // if row matches because of a global search, highlight that
-  if (cachedSearch.length > 0) {
+  if (clazz === "dataCell") {
+     // a little ugly: R deparses data cell values as list(col = val, col = val...).  when rendering
+     // a data cell which appears to have this format, count the assignment tokens to get a variable
+     // count and show that as a summary. 
+     if (data.substring(0, 5) === "list(" && data.indexOf("=") > 0) {
```

Comment:
If I'm correct, this code path is for `data.frame`s included as part of a cell. Does this handle `data.frame`s which themselves contain list-columns cleanly? E.g. what if you have:

```r
data <- list(alpha = list(beta = 1, gamma = 2, delta = 3))
df <- dplyr::as_data_frame(data)
data
```

More generally, what exactly is `data` here? Do we need to be careful about deparsing overly large lists / data frames?

Reply:
> Does this handle data.frames which themselves contain list-columns cleanly?

That's a good question. It depends on what you expect; the variables in the list columns will be included in the count. It's arguable whether or not that's what you'd expect, but I think it'll be rare in practice (to do this you need a list column inside a dataframe inside a dataframe) 

> More generally, what exactly is data here? Do we need to be careful about deparsing overly large lists / data frames?

`data` is the character form of this value:

https://github.com/rstudio/rstudio/blob/377bc9cf7dfdfa379ef72c67fcd3ff493acdde27/src/cpp/session/modules/data/DataViewer.cpp#L653-L659

This is existing behavior so if there were an issue with complicated/large values I think we'd have heard about it by now (although there's always room for improvement!)

Reply:
Got it! Since this PR isn't introducing any extra potential for trouble beyond what already exists I think that's fine.